HighwayPatrol.cpp

```diff
@@ -28,8 +32,29 @@ void HighwayPatrol::pullOver( Vehicle* v, bool willArrest, Highway* h )
     if( willArrest )
     {
         //print the vehicle type in this std::cout between "THE [" and "] PULL". 
-        std::cout << name << ": YOU IN THE [ " << " ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
+
+
+        std::cout << name << ": YOU IN THE [ " << getVehicleType(v) << " ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
         std::cout << "EVERYONE ELSE, SLOW DOWN!! \n\n\n";
         h->removeVehicle(v);
     }
 }
+
+std::string HighwayPatrol::getVehicleType( Vehicle* v) 
+{
+    std::string result = "vehicle";
+    if( auto* c = dynamic_cast<Car*>(v) ) { result = "Car"; }
+    if( auto* c = dynamic_cast<SemiTruck*>(v) ) { result = "Truck"; }
+    if( auto* c = dynamic_cast<Motorcycle*>(v) ) { result = "Motorcycle"; }
+    return result;
+}
+
+void HighwayPatrol::tryToEvade() 
+{
+     //never tries to evade 
+};
+
+void HighwayPatrol::setSpeed( int s ) 
+{
+     speed = s;
+};
```

Comment:
Remove these 2 functions.  they should not have been added.  The base class has an implementation. 
- [ ] remove functions.